TESTFQDNLookup failures

## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.7/detail/8.7/139//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.7/detail/8.7/139//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.7/detail/8.7/139//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.7/detail/8.7/139//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_null.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2023-05-22T16:20:50.503Z&rangeTo=2023-05-22T16:40:50.503Z&transactionName=BUILD+Beats%2Fbeats%2F8.7&transactionType=job&latencyAggregationType=avg&traceId=31dab8808cede7f3941c41aef3e25070&transactionId=bfc1181f2c4acba9)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-05-22T16:30:50.503+0000


* Duration: 65 min 34 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 2  |
| Passed       | 26093  |
| Skipped      | 1971 |
| Total        | 28066   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![2](https://img.shields.io/badge/2%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.7/detail/8.7/139//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `Build&Test / libbeat-goIntegTest / TestFQDNLookup/lookup_succeeds – github.com/elastic/beats/v7/libbeat/processors/add_host_metadata`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestFQDNLookup/lookup_succeeds
mockdns server: 2023/05/22 17:06:41 DNS TRACE ;; opcode: QUERY, status: NOERROR, id: 13882
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;beats-ci-immutable-ubuntu-2204-1684773823855146840.	IN	 A

;; ANSWER SECTION:
beats-ci-immutable-ubuntu-2204-1684773823855146840.	9999	IN	CNAME	foo.bar.baz.
foo.bar.baz.	9999	IN	A	1.1.1.1
mockdns server: 2023/05/22 17:06:41 DNS TRACE ;; opcode: QUERY, status: NOERROR, id: 10645
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;beats-ci-immutable-ubuntu-2204-1684773823855146840.	IN	 AAAA

;; ANSWER SECTION:
beats-ci-immutable-ubuntu-2204-1684773823855146840.	9999	IN	CNAME	foo.bar.baz.
    add_host_metadata_test.go:571: 
        	Error Trace:	/var/lib/jenkins/workspace/8.7-139-3e8e3ab6-9412-4acb-afe3-7d4995133ed4/src/github.com/elastic/beats/libbeat/processors/add_host_metadata/add_host_metadata_test.go:571
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestFQDNLookup/lookup_succeeds
    --- FAIL: TestFQDNLookup/lookup_succeeds (0.08s)
 
 ```
</p></details>
</ul>
    
##### `Build&Test / libbeat-goIntegTest / TestFQDNLookup – github.com/elastic/beats/v7/libbeat/processors/add_host_metadata`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestFQDNLookup
--- FAIL: TestFQDNLookup (0.16s)
 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![4](https://img.shields.io/badge/4%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.7/detail/8.7/139//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `libbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 6 min 47 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.7/runs/139/steps/12484/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `libbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 3 min 36 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.7/runs/139/steps/16617/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `libbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 3 min 37 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.7/runs/139/steps/18281/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.7/runs/139/steps/19512/log/?start=0">here</a></li>
<li>Description: <code>Error "hudson.AbortException: script returned exit code 1"</code></l1>
</ul>
  
  </p>
  </details>

--- a/libbeat/processors/add_host_metadata/add_host_metadata_test.go
+++ b/libbeat/processors/add_host_metadata/add_host_metadata_test.go
@@ -524,8 +524,8 @@ func TestFQDNLookup(t *testing.T) {
 		expectedFQDNLookupFailedCount int64
 	}{
 		"lookup_succeeds": {
-			cnameLookupResult:             "foo.bar.baz.",
-			expectedHostName:              "foo.bar.baz",
+			cnameLookupResult:             "example.com.",
+			expectedHostName:              "example.com",
 			expectedFQDNLookupFailedCount: 0,
 		},
 		"lookup_fails": {
@@ -569,6 +569,8 @@ func TestFQDNLookup(t *testing.T) {
 			addHostMetadataP, ok := p.(*addHostMetadata)
 			require.True(t, ok)
 			require.Equal(t, test.expectedFQDNLookupFailedCount, addHostMetadataP.metrics.FQDNLookupFailed.Get())
+			// reset so next run is correct, registry is global
+			addHostMetadataP.metrics.FQDNLookupFailed.Set(0)
 
 			// Run event through processor and check that hostname reported
 			// by processor is same as OS-reported hostname

